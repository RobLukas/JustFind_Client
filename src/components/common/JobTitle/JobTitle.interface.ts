interface JobTitleProps {
  children: string;
}

export default JobTitleProps;
