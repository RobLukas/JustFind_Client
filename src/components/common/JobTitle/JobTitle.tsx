import React, { FC } from 'react';
import JobTitleProps from './JobTitle.interface';
import S from './JobTitle.styles';

const JobTitle: FC<JobTitleProps> = ({ children }: JobTitleProps) => {
  return <S.JobTitle>{children}</S.JobTitle>;
};

export default JobTitle;
