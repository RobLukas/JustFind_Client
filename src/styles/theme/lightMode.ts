export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
