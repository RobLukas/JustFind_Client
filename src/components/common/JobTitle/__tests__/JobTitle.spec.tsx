import React from 'react';
import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';
import JobTitle from '../JobTitle';
import JobTitleProps from '../JobTitle.interface';
import { ThemeProvider } from 'styled-components';
import lightMode from 'styles/theme/lightMode';

describe('<JobTitle />', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const { children }: JobTitleProps = { children: 'job title' };
    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <JobTitle>{children}</JobTitle>
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
