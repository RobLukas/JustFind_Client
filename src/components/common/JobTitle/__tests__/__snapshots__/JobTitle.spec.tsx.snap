// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobTitle /> renders correctly 1`] = `
.c0 {
  color: #37474f;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 16px;
  margin: 0 5px 0 0;
  font-weight: 600;
  line-height: 23px;
}

@media (max-width:1020px) {
  .c0 {
    white-space: initial;
    overflow: initial;
    text-overflow: initial;
  }
}

<span
  class="c0"
>
  job title
</span>
`;
